cli/aiconfig-editor/pages/index.tsx

```diff
@@ -7,34 +8,56 @@ import { ufetch } from "ufetch";
 type EditorFile = {
   name: string;
   extension: string;
-  isFolder: boolean;
-  disabled: boolean;
+  path: string;
+  isDirectory: boolean;
+  disabled?: boolean;
 };
 
 export default function Home() {
   // Get list of files in the currect directory; this should also use a path from the router to become a nested file tree that can be navigated
   const router = useRouter();
 
-  const [path, setPath] = useState<undefined | string>(undefined);
   const [files, setFiles] = useState<EditorFile[]>([]);
 
   const getFiles = useCallback(async () => {
-    const res = await ufetch.post("/api/files", { path });
+    const res = await ufetch.post("/api/files", { path: router.query.path });
 
-    console.log(res);
-  }, [path]);
+    const newFiles = res.files.map((file: EditorFile) => {
+      const disabled = !file.isDirectory && file.extension !== ".json";
+
+      return {
+        ...file,
+        disabled,
+      };
+    });
+
+    setFiles(newFiles);
+  }, [router.query.path]);
 
   useEffect(() => {
     // Get files from current directory (will be whatever is passed as root directory when undefined, but needs to be handled server side
     getFiles();
   }, [getFiles]);
 
-  const navigate = useCallback(() => {
-    // If directory, then change path & also update the url route to add to history
-    // If random file, should not be able to select
-    // If aiconfig.json file, then open editor - can't actually detect yet, so will just try to parse
-    // load_json
-  }, []);
+  const navigate = useCallback(
+    (file: EditorFile) => {
+      // If directory, then change path & also update the url route to add to history
+      // If random file, should not be able to select
+      // If aiconfig.json file, then open editor - can't actually detect yet, so will just try to parse
```

Comment:
Ah, this is another place the load schema validation would help; https://github.com/lastmile-ai/aiconfig/issues/291